app.rb

```diff
@@ -0,0 +1,160 @@
+require './book'
+require './rental'
+require './person'
+require './student'
+require './teacher'
+
+class App
+  attr_accessor :books, :rentals, :people, :student, :teacher
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    person_type = person_type_input
+    if person_type == 1
+      person = create_student
+    elsif person_type == 2
+      person = create_teacher
+    else
+      puts 'Invalid input.'
+      return
+    end
+    push_person_to_list(person)
+    puts 'Person created successfully'
+  end
+
+  def person_type_input
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    gets.chomp.to_i
+  end
+
+  def create_student
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has Parent Permission? [Y/N]:'
+    parent_permission = gets.chomp.downcase == 'y'
+    Student.new(age, name, parent_permission)
+  end
+
+  # def create_student
+  #   student_data = gather_student_data
+  #   student = Student.new(student_data[:age], student_data[:name], student_data[:parent_permission])
+  #   push_person_to_list(student)
+  #   puts 'Person created successfully'
+  # end
+
+  # def gather_student_data
+  #   student_data = {}
+  #   puts 'Age:'
+  #   student_data[:age] = gets.chomp
+  #   puts 'Name:'
+  #   student_data[:name] = gets.chomp
+  #   puts 'Has Parent Permission? [Y/N]:'
+  #   student_data[:parent_permission] = gets.chomp.downcase == 'y'
+  #   student_data
+  # end
+
+  # def create_teacher
+  #   puts 'Age:'
+  #   age = gets.chomp
+  #   puts 'Name:'
+  #   name = gets.chomp
+  #   puts 'Specialization:'
+  #   specialization = gets.chomp
+  #   Teacher.new(age, name, specialization)
+  # end
```

Comment:
- [ ] Impressive progress in developing the UI of your school library app. I'd like to provide a constructive suggestion to enhance your codebase further. It's crucial to maintain a clean and organized codebase, which leads me to recommend the removal of any commented-out `code` within your files.

    - In pursuit of an optimized and well-maintained code, I urge you to undertake the task of removing all commented-out `code`. This proactive step will not only improve the present state of your code but will also simplify future development and collaboration.

Happy coding! 😊🚀🍀